tests/unit/models/gpflux/test_sampler.py

```diff
@@ -157,3 +172,370 @@ def test_deep_gaussian_process_sampler_samples_are_distinct_for_new_instances(
 
     xs = tf.random.uniform([100, 2], minval=-10.0, maxval=10.0, dtype=tf.float64)
     npt.assert_array_less(1e-9, tf.abs(sampler2.sample(xs) - sampler1.sample(xs)))
+
+
+@pytest.mark.parametrize("num_features", [0, -2])
+def test_dgp_decoupled_trajectory_sampler_raises_for_invalid_number_of_features(
+    num_features: int,
+    two_layer_model: Callable[[TensorType], DeepGP],
+    keras_float: None,
+) -> None:
+    _, model = _build_dataset_and_train_deep_gp(two_layer_model)
+
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        DeepGaussianProcessDecoupledTrajectorySampler(model, num_features=num_features)
+
+
+def test_dgp_decoupled_trajectory_sampler_raises_for_invalid_model(keras_float: None) -> None:
+    with pytest.raises(ValueError, match="Model must be .*"):
+        DeepGaussianProcessDecoupledTrajectorySampler(
+            QuadraticMeanAndRBFKernel(), 10  # type: ignore
+        )
+
+
+@pytest.mark.parametrize("num_evals", [10, 100])
+def test_dgp_decoupled_trajectory_sampler_returns_trajectory_function_with_correct_shapes(
+    num_evals: int,
+    two_layer_model: Callable[[TensorType], DeepGP],
+    keras_float: None,
+) -> None:
+    batch_size = 5
+
+    _, model = _build_dataset_and_train_deep_gp(two_layer_model)
+
+    sampler = DeepGaussianProcessDecoupledTrajectorySampler(model)
+
+    trajectory = sampler.get_trajectory()
+    xs = tf.random.uniform([num_evals, 2], minval=-10.0, maxval=10.0, dtype=tf.float64)  # [N, D]
+    xs_with_dummy_batch_dim = tf.expand_dims(xs, -2)  # [N, 1, D]
+    xs_with_full_batch_dim = tf.tile(xs_with_dummy_batch_dim, [1, batch_size, 1])  # [N, B, D]
```

Comment:
you use this snippet in many tests here, perhaps turn it into a private function somewhere at the top and call everywhere?